telegram/ext/picklepersistence.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python
+#
+# A library that provides a Python interface to the Telegram Bot API
+# Copyright (C) 2015-2018
+# Leandro Toledo de Souza <devs@python-telegram-bot.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains the BasePersistence class."""
+import pickle
+from collections import defaultdict
+
+from telegram.ext import BasePersistence
+
+
+class PicklePersistence(BasePersistence):
```

Comment:
Is there ever a file where it makes sense to use single_file=False?

Reply:
dunno, but it might

Reply:
I'm not really a fan of having features that we aren't sure anyone can use... It's just more maintenance work for us :/

Reply:
Also you can't choose the filetype (suffix) when using single_file=False.?

Reply:
No, but I don;t really see a problem here. I think it's versatile enough this way.